mirdata/billboard.py

```diff
@@ -0,0 +1,453 @@
+        self.chart_date = self._track_metadata["chart_date"]
+        self.target_rank = self._track_metadata["target_rank"]
+        self.actual_rank = self._track_metadata["actual_rank"]
+        self.title = self._track_metadata["title"]
+        self.artist = self._track_metadata["artist"]
+        self.peak_rank = self._track_metadata["peak_rank"]
+        self.weeks_on_chart = self._track_metadata["weeks_on_chart"]
+
+    def __repr__(self):
+        repr_string = (
+            "Billboard Track(track_id={}, audio_path={}, "
+            + "chart_date={}, "
+            + "title={}, artist={}, "
+            + "SectionData('intervals', 'labels')"
+        )
+        return repr_string.format(
+            self.track_id, self.audio_path, self.chart_date, self.title, self.artist
+        )
+
+    @utils.cached_property
+    def chords(self):
+        return {
+            "full": _load_chords(self.lab_full_path),
+            "majmin7": _load_chords(self.lab_majmin7_path),
+            "majmin7inv": _load_chords(self.lab_majmin7inv_path),
+            "majmin": _load_chords(self.lab_majmin_path),
+            "majmininv": _load_chords(self.lab_majmininv_path),
+        }
+
+    @utils.cached_property
+    def sections(self):
+        return _load_sections(
+            os.path.join(self._data_home, self._track_paths["salami"][0])
+        )
+
+    @property
+    def audio(self):
+        return librosa.load(self.audio_path, sr=None, mono=True)
+
+    def to_jams(self):
+        return jams_utils.jams_converter(
+            chord_data=[
+                (self.chords["full"], "full"),
+                (self.chords["majmin7"], "majmin7"),
+                (self.chords["majmin7inv"], "majmin7inv"),
+                (self.chords["majmin"], "majmin"),
+                (self.chords["majmininv"], "majmininv"),
+            ],
+            section_data=[(self.sections, None)],
+            metadata=self._track_metadata,
+        )
+
+
+def _load_chords(path):
+    """Private function to load LAB format chord data from a file
+
+    Args:
+        chords_path (str):
+
+    """
+
+    if path is None or not os.path.exists(path):
+        return None
+
+    start_times, end_times, chords = [], [], []
+    with open(path, "r") as f:
+        for l in f:
+            l = l.rstrip()
+            if l:
+                start, end, label = l.split("\t")
+                start_times.append(float(start))
+                end_times.append(float(end))
+                chords.append(label)
+
+    chord_data = utils.ChordData(np.array([start_times, end_times]).T, chords)
+    return chord_data
+
+
+def _load_sections(sections_path):
+    """Private function to load SALAMI format sections data from a file
+
+    Args:
+        sections_path (str):
+
+    """
+    if sections_path is None or not os.path.exists(sections_path):
+        return None
+
+    timed_sections = _timed_sections(_parse_salami(sections_path))
```

Comment:
Reading the code down below I'm a bit mixed if these are sections or chords, @Spijkervet can you tell me a bit about it?